Mathlib/Tactic/SlimCheck.lean

```diff
@@ -177,22 +177,23 @@ elab_rules : tactic | `(tactic| slim_check $[$cfg]?) => withMainContext do
       || (← isTracingEnabledFor `slim_check.shrink.candidates) }
   let inst ← try
     synthInstance (← mkAppM ``Testable #[tgt'])
-  catch _ => throwError "Failed to create a `testable` instance for `{tgt}`.
-What to do:
-1. make sure that the types you are using have `SlimCheck.SampleableExt` instances
-   (you can use `#sample my_type` if you are unsure);
-2. make sure that the relations and predicates that your proposition use are decidable;
-3. make sure that instances of `SlimCheck.Testable` exist that, when combined,
-   apply to your decorated proposition:
-```
-{tgt'}
-```
-
-Use `set_option trace.Meta.synthInstance true` to understand what instances are missing.
-
-Try this:
-set_option trace.Meta.synthInstance true
-#synth SlimCheck.Testable ({tgt'})"
+  catch _ => throwError "\
+      Failed to create a `testable` instance for `{tgt}`.\
+    \nWhat to do:\
+    \n1. make sure that the types you are using have `SlimCheck.SampleableExt` instances\
+    \n  (you can use `#sample my_type` if you are unsure);\
+    \n2. make sure that the relations and predicates that your proposition use are decidable;\
+    \n3. make sure that instances of `SlimCheck.Testable` exist that, when combined,\
+    \n  apply to your decorated proposition:\
+    \n```\
+    \n{tgt'}\
+    \n```\
+    \n\
+    \nUse `set_option trace.Meta.synthInstance true` to understand what instances are missing.\
+    \n\
+    \nTry this:\
+    \nset_option trace.Meta.synthInstance true\
+    \n#synth SlimCheck.Testable ({tgt'})"
```

Comment:
The reason this is here is because fixing  `#help` caused the set of hash commands used by `lint_hash_commands` to include `#synth`, which now causes a false positive on this line. I fixed it by indenting the string literal so it doesn't look like an actual `#synth` call.

Reply:
I think #13142 should fix this: the new Lean version of this linter ought to see this is a string literal.

Reply:
I just ran an experiment: #13230 shows that building mathlib succeeds (in particular, the syntax linter does not flag `#synth`), but the text-based one works.

I think that the `#`-command CI checks should be removed.

Reply:
I don't understand what you are trying to say. Prior to this commit, `#help` would not flag `#synth` as a hash command, which causes the text-based CI to produce false negatives for all `#synth` commands. After it, `#synth` is treated as a hash command, producing a false positive right here in a string literal. I think the string literal is better anyway with proper indentation. The new syntax linter does not use `#help` to begin with and just fires on all atoms starting with `#` (which has its own set of false negatives since it doesn't recurse into subexpressions correctly, basically reproducing the error fixed in this PR in a new implementation, in addition to the FN because it doesn't consider nested commands like `set_option ... in #synth`).

Reply:
What I was trying to say is that it is only the string-based, `#help`-dependent linter that flags this, not the more recent syntax linter.  Hence, removing the older string-based linter would make this change on Slimcheck not needed.  Of course, the change could still be desirable!